packages/dashboard/src/components/elements/account-wizard/account-wizard.tsx

```diff
@@ -0,0 +1,284 @@
+                        })}
+                        onClick={() => navigateTo(i)}
+                    >
+                        <Icon
+                            size={24}
+                            icon={i < current_step ? 'IcDashboardCheck' : item.icon}
+                            custom_color={i < current_step ? 'var(--brand-red-coral)' : ''}
+                            color={isDisabled(i) ? 'disabled' : ''}
+                        />
+                        <DesktopWrapper>
+                            <Text
+                                size='xxs'
+                                weight={i === current_step ? 'bold' : ''}
+                                className='dw-account-wizard__header-step-title'
+                                color={isDisabled(i) ? 'disabled' : ''}
+                            >
+                                {i + 1}. {item.header.title}
+                            </Text>
+                            <div className='dw-account-wizard__header-step-line' />
+                        </DesktopWrapper>
+                        <MobileWrapper>
+                            <div className='dw-account-wizard__header-step-title-wrapper'>
+                                <Text size='xxs' className='dw-account-wizard__header-step-title'>
+                                    {localize('Step {{current_step}} of {{total_step}}', {
+                                        current_step,
+                                        total_step: items.length,
+                                    })}
+                                </Text>
+                                <Text size='s' weight='bold' className='dw-account-wizard__header-step-title'>
+                                    {item.header.title}
+                                </Text>
+                            </div>
+                        </MobileWrapper>
+                    </div>
+                ))}
+            </div>
+            <div className='dw-account-wizard__header-active-title'>
+                <Text size='m' weight='bold' styles={{ lineHeight: '3.6rem' }}>
+                    {items[current_step]?.header.active_title}
+                </Text>
+            </div>
+        </div>
+    );
+};
+
+type TAccountWizard = {
+    fetchResidenceList?: () => void;
+    fetchStatesList?: () => void;
+    fetchFinancialAssessment?: () => void;
+    needs_financial_assessment?: () => boolean;
+    has_currency?: () => boolean;
+    has_real_account?: () => boolean;
+    account_settings?: GetSettings;
+    [x: string]: any;
+};
+
+const AccountWizard: React.FC<TAccountWizard> = observer((props: TAccountWizard) => {
+    const [form_error, setFormError] = React.useState('');
+    const [previous_data, setPreviousData] = React.useState([]);
+    const [state_items, setStateItems] = React.useState<TWizardItemConfig[]>([]);
+    const [has_previous_data, setHasPreviousData] = React.useState(false);
+    const [mounted, setMounted] = React.useState(false);
```

Comment:
There's a shared hook for this here:

- `packages/shared/src/utils/hooks/mounted.js`